Tests/PackageModelSyntaxTests/ManifestEditTests.swift

```diff
@@ -0,0 +1,312 @@
+                ]
+            )
+            """, expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                  .package(url: "https://github.com/apple/swift-syntax.git", from: "510.0.1"),
+                  .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ]
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+
+    func testAddPackageDependencyExistingEmpty() {
+        assertManifestRefactor("""
+            let package = Package(
+                name: "packages",
+                dependencies: [ ]
+            )
+            """,
+            expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                    .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ]
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+
+    func testAddPackageDependencyNoExistingAtEnd() {
+        assertManifestRefactor("""
+            let package = Package(
+                name: "packages"
+            )
+            """, 
+            expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                    .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ]
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+
+    func testAddPackageDependencyNoExistingMiddle() {
+        assertManifestRefactor("""
+            let package = Package(
+                name: "packages",
+                targets: []
+            )
+            """,
+            expectedManifest: """
+            let package = Package(
+                name: "packages",
+                dependencies: [
+                    .package(url: "https://github.com/apple/swift-system.git", .branch("main")),
+                ],
+                targets: []
+            )
+            """) { manifest in
+            AddPackageDependency.addPackageDependency(
+                Self.swiftSystemPackageDependency,
+                to: manifest,
+                manifestDirectory: try! AbsolutePath(validating: "/")
+            )
+        }
+    }
+}
```

Comment:
I think it would be good to also have a test case where there is an existing dependency on a single line array literal. Ie add a dependency to the following

```swift
let package = Package(
    name: "packages",
    dependencies: [.package(url: "https://github.com/apple/swift-syntax.git", from: "510.0.1")]
)
```

Reply:
... and it uncovered an issue with trivia, thanks!